cmake/templates/oneDPLConfig.cmake.in

```diff
@@ -18,6 +18,16 @@ get_filename_component(_onedpl_root "${_onedpl_root}/../../../" ABSOLUTE)
 @HANDLE_HEADERS_PATH@
 if (EXISTS "${_onedpl_headers}")
     if (NOT TARGET oneDPL)
+        if (CMAKE_CXX_COMPILER MATCHES ".*(dpcpp-cl|icx-cl|icx)(.exe)?$")
+            set(INTEL_LLVM_COMPILER_MSVC_LIKE TRUE)
+        elseif(CMAKE_CXX_COMPILER MATCHES ".*(dpcpp|icpx)(.exe)?$")
+            set(INTEL_LLVM_COMPILER_GNU_LIKE TRUE)
+        endif()
+
+        if ((NOT WindowsIntelLLVM_FOUND) AND CMAKE_HOST_WIN32 AND (INTEL_LLVM_COMPILER_MSVC_LIKE OR INTEL_LLVM_COMPILER_GNU_LIKE))
+            message(WARNING "On Windows, ${CMAKE_CXX_COMPILER} requires some workarounds to function properly with CMake. We recommend using 'find_package(WindowsIntelLLVM)' before your 'project()' call to apply these workarounds.")
+        endif()
```

Comment:
It does not look right to me to always show this warning on Windows. It is probably better to show it in more specific cases and leave a link to readme for more details.

Reply:
This warning is shown on Windows when the workaround package `WindowsIntelLLVM` has not been included first and we are using an intel compiler which requires the workarounds to function.  

This is the situation where we may see errors in generating the build scripts or in compiling or linking.  Once the workaround package is included, this is no longer shown to the user.

I don't think that we can narrow it down much further for CMake version, unless the newly released 3.26.x works fully on one or both of the compilers without workarounds.  I'll look into if this is the case.